exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,187 @@
+        assignCategoriesToBoardSpaces();
+    }
+
+	public void addPlayer(String playerName) {
+	    allPlayers.add(playerName);
+	    locationOnBoard[numberOfPlayers()] = BOARD_STARTING_LOCATION;
+	    goldCoins[numberOfPlayers()] = ZERO_COINS;
+	    isInPenaltyBox[numberOfPlayers()] = false;
+
+	    System.out.println(playerName + " was added");
+	    System.out.println("They are player number " + numberOfPlayers());
+    }
+
+	public void rollDice(int currentRoll) {
+		System.out.println(allPlayers.get(currentPlayer) + " is the current player");
+		System.out.println("They have rolled a " + currentRoll);
+
+		if (isInPenaltyBox[currentPlayer]) {
+            penaltyTakeTurn(currentRoll);
+        } else {
+            takeTurn(currentRoll);
+        }
+	}
+
+    private void penaltyTakeTurn(int currentRoll) {
+        boolean oddRoll = currentRoll % 2 != 0;
+        if (oddRoll) {
+            isExitingPenaltyBox = true;
+            System.out.println(allPlayers.get(currentPlayer) + " is getting out of the penalty box");
+            takeTurn(currentRoll);
+        } else {
+            System.out.println(allPlayers.get(currentPlayer) + " is not getting out of the penalty box");
+            isExitingPenaltyBox = false;
+        }
+    }
+
+    public boolean isCorrectlyAnswered() {
+        if (isInPenaltyBox[currentPlayer]){
+            return penaltyCorrectAnswer();
+        }
+        else {
+            System.out.println("Answer was corrent!!!!");
+            collectCoins();
+            return hasCurrentPlayerWon();
+        }
+    }
+
+    private boolean penaltyCorrectAnswer() {
+        if (isExitingPenaltyBox) {
+            System.out.println("Answer was correct!!!!");
+            collectCoins();
+            return hasCurrentPlayerWon();
+        } else {
+            getNextPlayer();
+            return true;
+        }
+    }
+
+    public boolean isIncorrectlyAnswered(){
+        System.out.println("Question was incorrectly answered");
+        System.out.println(allPlayers.get(currentPlayer)+ " was sent to the penalty box");
+        isInPenaltyBox[currentPlayer] = true;
+
+        getNextPlayer();
+        return true;
+    }
+
+    private void takeTurn(int currentRoll) {
+        locationOnBoard[currentPlayer] = locationOnBoard[currentPlayer] + currentRoll;
+        int currentLocation = locationOnBoard[currentPlayer];
+        if (currentLocation > END_SPACE) {
+            locationOnBoard[currentPlayer] = currentLocation - NUMBER_OF_SPACES;
+        }
+
+        System.out.println(allPlayers.get(currentPlayer)
+                + "'s new location is "
+                + locationOnBoard[currentPlayer]);
+        System.out.println("The category is " + getCurrentCategory());
+
+        askQuestion();
+    }
+
+    private int numberOfPlayers() {
+        return allPlayers.size();
+    }
+
+    private void askQuestion() {
+		if (getCurrentCategory().equals(POP))
+			System.out.println(popQuestions.removeFirst());
```

Comment:
Not sure if this is a Java think but this removing of a question and printing it in the same line is NICE! 🙌